lib/places_alloverse_com/accounts/credential.ex

```diff
@@ -17,4 +21,105 @@ defmodule PlacesAlloverseCom.Accounts.Credential do
     |> validate_required([:email])
     |> unique_constraint(:email)
   end
+
+  @doc """
+    A user changeset for registration.
+    It is important to validate the length of both e-mail and password.
+    Otherwise databases may truncate the e-mail without warnings, which
+    could lead to unpredictable or insecure behaviour. Long passwords may
+    also be very expensive to hash for certain algorithms.
+  """
+
+  def registration_changeset(user, attrs) do
+    user
+    |> cast(attrs, [:email, :password])
+    |> validate_email()
+    |> validate_password()
+  end
+
+  defp validate_email(changeset) do
+    changeset
+    |> validate_required([:email])
+    |> validate_format(:email, ~r/^[^\s]+@[^\s]+$/, message: "must have the @ sign and no spaces")
+    |> validate_length(:email, max: 160)
+    |> unsafe_validate_unique(:email, PlacesAlloverseCom.Repo)
+    |> unique_constraint(:email)
+  end
+
+  defp validate_password(changeset) do
+    changeset
+    |> validate_required([:password])
+    |> validate_length(:password, min: 8, max: 80)
+    # |> validate_format(:password, ~r/[a-z]/, message: "at least one lower case character")
+    # |> validate_format(:password, ~r/[A-Z]/, message: "at least one upper case character")
+    # |> validate_format(:password, ~r/[!?@#$%^&*_0-9]/, message: "at least one digit or punctuation character")
```

Comment:
I guess nobody wants these so we can remove these three commented-out  altogether? :sweat_smile: